source/common/filter/tcp_proxy.cc

```diff
@@ -160,6 +160,7 @@ void TcpProxy::initializeReadFilterCallbacks(Network::ReadFilterCallbacks& callb
   ENVOY_CONN_LOG(debug, "new tcp proxy session", read_callbacks_->connection());
 
   read_callbacks_->connection().addConnectionCallbacks(downstream_callbacks_);
+  read_callbacks_->connection().enableHalfClose(true);
```

Comment:
I'm not convinced we want this to be the default without any option. I'm wondering if this should be an opt-in option on the filter? WDYT?

Reply:
I think properly proxying half-closes should at least be the default.  Is there any concrete use-case where this behavior causes problems?  In general, if either the client or the server can't deal with half-close semantics, they'll signal that (via a RST), so the proxy shouldn't need to modify TCP behavior.

Reply:
I don't know. I'm mainly concerned that this is not the current behavior and something will break.

Reply:
I thought about it, and I agree it sounds reasonable to just do half-close processing for TCP proxy by default.